File: Venda-distribuida-java/src/main/java/br/ufc/venda/client/proxy/ProdutoProxy.java
package br.ufc.venda.client.proxy;

import java.util.List;

import br.ufc.venda.client.dao.ProdutoDAO;
import br.ufc.venda.model.Produto;

public class ProdutoProxy {
	
	ProdutoDAO pdao;
	
	public ProdutoProxy(){
		pdao = new ProdutoDAO();
	}
	
	public List<Produto> listarProdutos(){
		return pdao.selectAll();
	}
	
	public List<Produto> encontrarProduto(int id){
		return pdao.select(id);
	}
	
}

File: Venda-distribuida-python/proxy/ProdutoProxy.py
#-*-coding:utf-8 -*-

from dao.ProdutoDAO import ProdutoDAO

class ProdutoProxy():

    def __init__(self):
        self.pdao = ProdutoDAO()

    def listarProdutos(self):
        return self.pdao.selectAll()

    def encontrarProduto(self, id):
        return self.pdao.select(id)

